### PR TITLE
Fixes #24545: Allow shared iOS builds

### LINKED ISSUE #24545: Shared library should be allowed for iOS
Hi,

We're trying to switch from our CMake based build system to openssl's native build system. We build openssl on several platforms and currently I see a small issue.

Shared openssl for iOS was disabled with this commit: https://github.com/openssl/openssl/commit/aa58071e4b8b245db1564f476731c978738e7e98

The description claims that "iOS doesn't have the concept of shared libraries", what is wrong. One may build shared library and use it as part of a framework. Also dependant frameworks may reuse shared libraries from framework on which it depends.

### PR DESCRIPTION
**Pull request by fwh-dc**
Fixes #24545

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: Configurations/15-ios.conf [hunk 1]
```diff
@@ -11,7 +11,7 @@ my %targets = (
         template         => 1,
         inherit_from     => [ "darwin-common" ],
         sys_id           => "iOS",
-        disable          => [ "shared", "async" ],
+        disable          => [ "async" ],
```
Comment (nhorman) [inline Configurations/15-ios.conf:14]: IOS is an unadopted platform.  As such CI doesn't test this and I (personally) have no way to verify it.  Can you offer some evidence that this works properly?

Comment (fwh-dc) [reply Configurations/15-ios.conf:14]: I did try it locally and it produces a dylib. Are there other tests I can do? Would it be possible to add a ci job to test ios builds?

Comment (Sashan) [reply Configurations/15-ios.conf:14]: the change looks good. one possible test worth to try is to check if  'make install' drops the dynamic library to install prefix. there is not much else we can do I think.

Comment (fwh-dc) [reply Configurations/15-ios.conf:14]: I did a test as requested and it works as described.

**Review by Sashan — APPROVED** [review]
looks good to me, I think this should go in.

**Review by tmshort — APPROVED** [review]

**Comment by t8m** [comment]
Merged to all the active branches. Thank you for your contribution.

**Pull request closed without merging**